app/src/main/java/com/gusta/wakemehome/GoogleMaps.java

```diff
@@ -0,0 +1,307 @@
+package com.gusta.wakemehome;
+
+import android.content.DialogInterface;
+import android.content.pm.PackageManager;
+
+import androidx.annotation.NonNull;
+
+import android.graphics.Color;
+import android.location.Geocoder;
+import android.location.Location;
+
+import androidx.core.app.ActivityCompat;
+import androidx.core.content.ContextCompat;
+
+import android.util.Log;
+
+import com.google.android.gms.common.api.Status;
+import com.google.android.gms.location.FusedLocationProviderClient;
+import com.google.android.gms.location.LocationServices;
+import com.google.android.gms.maps.CameraUpdateFactory;
+import com.google.android.gms.maps.GoogleMap;
+import com.google.android.gms.maps.OnMapReadyCallback;
+import com.google.android.gms.maps.SupportMapFragment;
+import com.google.android.gms.maps.model.Circle;
+import com.google.android.gms.maps.model.CircleOptions;
+import com.google.android.gms.maps.model.LatLng;
+import com.google.android.gms.maps.model.MarkerOptions;
+import com.google.android.gms.tasks.OnCompleteListener;
+import com.google.android.gms.tasks.Task;
+import com.google.android.libraries.places.api.Places;
+
+import com.google.android.libraries.places.api.model.Place;
+import com.google.android.libraries.places.api.net.PlacesClient;
+import com.google.android.libraries.places.widget.AutocompleteSupportFragment;
+import com.google.android.libraries.places.widget.listener.PlaceSelectionListener;
+
+import java.util.Arrays;
+import java.util.Locale;
+
+import static com.gusta.wakemehome.MapAddress.getCoordinatesAddress;
+
+
+public class GoogleMaps extends MapProvider implements OnMapReadyCallback {
+
+    private static final String TAG = MapsActivity.class.getSimpleName();
+    private static final int circleWidth = 6;
+    private static final int circleDividorScale = 600;
+    private GoogleMap mMap;
+    private Geocoder mGeocoder;
+    private Circle mMapRadiusCircle;
+
+    // The entry point to the Fused Location Provider.
+    private FusedLocationProviderClient mFusedLocationProviderClient;
+
+
+    private static final int PERMISSIONS_REQUEST_ACCESS_FINE_LOCATION = 1;
+    private boolean mLocationPermissionGranted;
+
+    // The geographical location where the device is currently located. That is, the last-known
+    // location retrieved by the Fused Location Provider.
+    private Location mLastKnownLocation;
+
+    public GoogleMaps(MapsActivity mapsActivity){
```

Comment:
It's not weird at all, think about the meaning of GoogleMap class.
It will be a part of map activity, the main purpose is to be able to make multiple providers and only one Map activity.